Titko Vadim

Vadbeg opened the pull request:
Vadim Titko

email: Vadbeg@tut.by

Nickolai-Belov committed:
add readme

Vadbeg committed:
Method for getting rss from url is created

Vadbeg committed:
Methods for news ouput, image reading
and description reading were added

Every method was wrapped into class

Vadbeg committed:
Simple argparse manipulations were added.

New features:
	- from dict to json method created
	- news format was slightly changed

Vadbeg committed:
Setuptools usage was added. As a result were created pacakge
intsallation files (*.whl).

Because of that project structure changed. rss_reader.py was moved to
news_feed directory.

Vadbeg committed:
News cashing was added to method _cash_news

Vadbeg committed:
News cashing and reading by date were added

Vadbeg committed:
News converter (format_converter.py) was added. Now there
is just converter into pdf format using fpdf library

Vadbeg committed:
New converter into html was added.

Vadbeg committed:
Simple unittests where added.
Minor expressions where catched.

Vadbeg committed:
News cashing without news repeating
was added using pandas library

Vadbeg committed:
.gitignore was added

Vadbeg committed:
Merge branch 'master' of https://github.com/Vadbeg/PythonHomework

AlexeiBuzuma commented:
Это полностью оконечная работа? Если нет, то к имени пулл реквеста нужно добавить [WIP]

AlexeiBuzuma commented on .idea/inspectionProfiles/profiles_settings.xml:
```diff
@@ -0,0 +1,6 @@
+    <option name="USE_PROJECT_PROFILE" value="false" />
+    <version value="1.0" />
+  </settings>
+</component>
```
Не смотря на то, что папка .idea добавлена в gitignore, в репозитории все равно есть эти файлы. 
Нужно их удалить.

Vadbeg replied:
Удалил, спасибо.

AlexeiBuzuma commented on README:
```diff
@@ -0,0 +1,15 @@
+git #2 homework
```
Насколько я понял, это часть домашнего задания по гиту. Почему это в этом пулл реквесте? что-то попало лишнее сюда

Vadbeg replied:
Да, случайно получилось. Просто удалить тот README?

AlexeiBuzuma replied:
`pyc`  файлы еще есть в репозитории, их тоже лучше не хранить

AlexeiBuzuma commented on news_feed.egg-info/PKG-INFO:
```diff
@@ -0,0 +1,14 @@
+Metadata-Version: 2.1
```
Все файлы, которые относятся уже к созданному пакету (бинарники, папка dist/ и т.д.) тоже не стоит хранить в репозитории

Vadbeg replied:
Убрал всё, кроме того, что нужно для django.

Vadbeg committed:
SQLite usage was added

Vadbeg committed:
SQLite usage was added. Methods for caching into
sql database and reading from it were added.

Vadbeg committed:
Logging library usage was added. Argparse was modified.

Vadbeg committed:
Colorizetion was added (using colorama module). And main script exection method was rebuilded
and divided into several methods (depending on their independency)

Vadbeg committed:
Remove unuseful directories

Vadbeg committed:
.idea/ folder was removed

Vadbeg committed:
newsfeed django application core was created.

Several views were created:
	* rss_source - for getting rss link
	* index - for starting page
	* PostListView - for news page

Vadbeg committed:
Adding news into database was speed up.

Html templates were added (For all views).

Vadbeg committed:
News group by methods were added.

There is:
	* grouping by date
	* grouping by rss-source

Vadbeg committed:
Search tab was added. Now you can search news by date, description etc.

Vadbeg committed:
6th iteration:
	* pdf converter was added. Either into backend and frontend
	* news removing from database was added.

Vadbeg committed:
Bug with cyrillic letters was fixed:
	* fonts now aren't downloaded, they are on disk

Now page at which you are looking
transforms into pdf by clicking on button.
It works using global variable.
Another version with new table
in database is too slow.

Vadbeg committed:
Pdf converter now use not a global variable to read news, but:
PdfView class get names of news from html template and use
them to get rows which we need.

All views except rss source view use the same home.html template by now.

Vadbeg committed:
Pictures of rss sources were added

Vadbeg committed:
Bug with pagination during search was fixed

Before, you can't paginate through search result.

Vadbeg committed:
Bug with cyrillic letters was fixed for news_feed app.
Font for Russian language was added on disk.

Vadbeg committed:
RSS to Fb2 conversion class was added.

Vadbeg committed:
Rss news into fb2 conversion was added in Django app.

AlexeiBuzuma commented on setup.py:
```diff
@@ -0,0 +1,34 @@
+        ]
+    },
+
+    install_requires=['lxml>=4.3.0', 'requests', 'pandas'],
```
А зачем тут нужен pandas? :)

Vadbeg replied:
Использовал для кеширования новостей. Потом понял, что неправ и добавил SQLite...

AlexeiBuzuma commented on setup.py:
```diff
@@ -0,0 +1,34 @@
+    long_description = file.read()
+
+
+setup(
```
Кое-чего не хватает при установке в чистый контейнер 
```bash
# rss_reader --help
Traceback (most recent call last):
  File "/usr/local/bin/rss_reader", line 5, in <module>
    from news_feed.rss_reader import main
  File "/usr/local/lib/python3.8/site-packages/news_feed/rss_reader.py", line 30, in <module>
    from colorama import Fore, Style
ModuleNotFoundError: No module named 'colorama'
```

AlexeiBuzuma replied:
после установки колорамы вот такая ошибка 
```bash
# rss_reader --help
Traceback (most recent call last):
  File "/usr/local/bin/rss_reader", line 5, in <module>
    from news_feed.rss_reader import main
  File "/usr/local/lib/python3.8/site-packages/news_feed/rss_reader.py", line 32, in <module>
    from format_converter import (PdfNewsConverter,
ModuleNotFoundError: No module named 'format_converter'
```

Vadbeg replied:
Добавил нужные пакеты. С format_converter'ом поправил, спасибо.

AlexeiBuzuma commented on news_feed/rss_reader_unittest.py:
```diff
@@ -0,0 +1,54 @@
+import unittest
```
Есть некоторые стилистические ошибки в проекте 
```bash
# pycodestyle --max-line-length=120 .
./setup.py:34:1: W391 blank line at end of file
./newsfeed/news/render.py:54:1: W391 blank line at end of file
./newsfeed/news/views.py:133:82: E262 inline comment should start with '# '
./news_feed/format_converter.py:197:46: W291 trailing whitespace
./news_feed/format_converter.py:198:1: W293 blank line contains whitespace
./news_feed/format_converter.py:200:1: W293 blank line contains whitespace
./news_feed/format_converter.py:203:1: W293 blank line contains whitespace
./news_feed/format_converter.py:478:1: W391 blank line at end of file
./news_feed/rss_reader.py:197:27: W291 trailing whitespace
./news_feed/rss_reader.py:212:103: W291 trailing whitespace
./news_feed/rss_reader_unittest.py:12:121: E501 line too long (387 > 120 characters)
./news_feed/rss_reader_unittest.py:13:121: E501 line too long (326 > 120 characters)
./news_feed/rss_reader_unittest.py:14:121: E501 line too long (352 > 120 characters)
./news_feed/rss_reader_unittest.py:15:121: E501 line too long (698 > 120 characters)
./news_feed/rss_reader_unittest.py:16:121: E501 line too long (539 > 120 characters)
./news_feed/rss_reader_unittest.py:17:121: E501 line too long (533 > 120 characters)
./news_feed/rss_reader_unittest.py:18:121: E501 line too long (491 > 120 characters)
./news_feed/rss_reader_unittest.py:19:121: E501 line too long (511 > 120 characters)
./news_feed/rss_reader_unittest.py:20:121: E501 line too long (493 > 120 characters)
./news_feed/rss_reader_unittest.py:21:121: E501 line too long (572 > 120 characters)
```

AlexeiBuzuma commented:
readme файл пустой, нет информации с тем, как можно пользоваться этим приложением. при установке через `pip install . ` утилита корректно не работает
на данный момент у меня нет возможность проверить валидность работы решения

Vadbeg committed:
Package installation now is permitted.
Font files where added into package directory.

Vadbeg committed:
Bug with to-pdf view was fixed. Now news for pdf are finded by hash of title

Vadbeg committed:
mend

Vadbeg committed:
README.md was added

Vadbeg committed:
Bug with bad encoding in html was fixed

Vadbeg committed:
New unittests where added. Now mock is used for requests library.

Vadbeg committed:
Test section was added in README.md

Vadbeg committed:
Bug with view for fb2 downloading in Django was fixed. (news were finded by not appropriate column)

Vadbeg committed:
Python-dateutil library was added into setup.py

Vadbeg committed:
Bug with unsupported image files in pdf conversion was solved

Vadbeg committed:
UTF-8 encoding was added to json

Vadbeg committed:
Bug with colorama was fixed. Not it works on any OS

Vadbeg committed:
Readme was upgraded

Vadbeg committed:
Migrations were added

Vadbeg committed:
media_root directory now is created if web app needs it

Vadbeg committed:
get_news method was divided. Comments were upgraded

Vadbeg committed:
Problem with cyrillic letters in pdf and fb2 was ssolved.
Now I reuse pdf method from format_converter
